packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -201,22 +209,36 @@ contract MetaTransactionWallet is
    * @param values The CELO value to be sent with each transaction.
    * @param data The concatenated data to be sent in each transaction.
    * @param dataLengths The length of each transaction's data.
+   * @return The return values of all transactions appended as bytes and an array of the length
+   *         of each transaction output which will be 0 if a transaction had no output 
    */
   function executeTransactions(
     address[] calldata destinations,
     uint256[] calldata values,
     bytes calldata data,
     uint256[] calldata dataLengths
-  ) external {
+  ) external returns (bytes memory, uint256[] memory) {
     require(
       destinations.length == values.length && values.length == dataLengths.length,
       "Input arrays must be same length"
     );
+
+    bytes memory returnValues;
+    uint256[] memory returnLengths = new uint256[](destinations.length);
     uint256 dataPosition = 0;
     for (uint256 i = 0; i < destinations.length; i = i.add(1)) {
-      executeTransaction(destinations[i], values[i], sliceData(data, dataPosition, dataLengths[i]));
+      bytes memory returnVal = executeTransaction(
+        destinations[i],
+        values[i],
+        sliceData(data, dataPosition, dataLengths[i])
+      );
+      returnValues = abi.encodePacked(returnValues, returnVal);
```

Comment:
this is basically an append right?
would it be useful to ensure that `returnValues.length before == returnValues.length after - returnVal.length` as we expect?

Reply:
I'm not understanding this question. the length of `returnValues` is pretty unpredictable since some transactions do not return anything